### DISCUSSION TITLE
How to profile query time per request context?

### DISCUSSION THREAD

**Discussion by l-bujakowski** [mikro-orm/mikro-orm#1409] [Q&A]
Hello, I have a question.

What I am trying to achieve is to collect total time spent on db communication per http request.
I'm using EntityManager.fork(), to isolate between http requests.  
I've read in docs, that a custom logger may be configured, but its isntace is shared between all entity mangers. Just like they share driver and connection instances.

Any ideas, where I could possible hook, to collect time for all kind of db communication made by mikro-orm through given entity manager fork?

**Comment by B4nan** [comment]
As you mentioned, it's sinleton, just like connection and driver classes, I don't see any direct way to do so.

You could use flush events if you are fine with measuring the whole flush operation instead of queries:

https://mikro-orm.io/docs/lifecycle-hooks/#flush-events

Or you could probably leverage domain api or `AsyncLocalStorage` - you could store the request id from a middleware and access it from the logger callback.
